### PR TITLE
[docs] add tutorial on latency

### LINKED ISSUE #1560: Document latency tips
- What programming styles should we uses/avoid
- How do we measure it
- What techniques have we tried to improve things

### PR DESCRIPTION
**Pull request by odow**
Closes #1560

### CODE REVIEW

File: docs/src/tutorials/latency.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        inner::T
+    end
+    optimize!(x::Wrapper) = optimize!(x.inner)
+end
+
+module MyOptimizer
+    using ..MyMOI
+    struct Optimizer end
+    MyMOI.optimize!(x::Optimizer) = 1
+end
```
Comment (matbesancon) [inline docs/src/tutorials/latency.md:89]: Code inside modules is usually not indented
```suggestion
module MyMOI
struct Wrapper{T}
    inner::T
end
optimize!(x::Wrapper) = optimize!(x.inner)
end

module MyOptimizer
using ..MyMOI
struct Optimizer end
MyMOI.optimize!(x::Optimizer) = 1
end
```

**Comment by matbesancon** [comment]
I would add pointers to the JuliaCon workshop and talk on latency:
https://www.youtube.com/watch?v=rVBgrWYKLHY
https://www.youtube.com/watch?v=wXRMwJdEjX4
and the corresponding packages

**Review by frapac — COMMENTED** [review]
This is an interesting read. Just found out two typos.

File: docs/src/tutorials/latency.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+Lack of method ownership happens when a call is made using a mix of structs and
+methods from different modules. Because of this, no single module "owns" the
+method that is being dispatched, and so it cannot be precompiled.
+
+!!! tip
+    This is a slightly simplified explanation. Read the [precompilation tutorial](https://julialang.org/blog/2021/01/precompile_tutorial/)
+    for a more in-depth discussion on back-edges.
+
+Unfortunately, the design of MOI means that this is a frequent occurence! We
```
Comment (dourouc05) [inline docs/src/tutorials/latency.md:71]: occurrence

**Comment by odow** [comment]
Merged. We can improve in a future PR.

**Pull request merged**